components/dashboard/src/projects/project-context.tsx

```diff
@@ -110,7 +112,7 @@ export function useCurrentProject(): { project: Project | undefined; loading: bo
             setProject(project);
             setLoading(false);
         })();
-    }, [setProject, org.data, user, orgs.data, location, history, projectInfo]);
+    }, [setProject, org.data, user, orgs.data, location, history, projectInfo, listProjects.data]);
 
     return { project, loading };
```

Comment:
Should remove the `loading` state in this hook and just return `loading: isLoading` here (where isLoading is from the `useListProjectsQuery()`

Reply:
@selfcontained, I'm afraid this has some unwanted impact. When trying to load any project's page, e.g. https://at-enablep552d678423.preview.gitpod-dev.com/projects/bel-5a6d3595-5c81-4667-ab79-795b7688e817, or selecting a project from the project cards view, I'm returning back to the project cards view. I put a breakpoint here and see that `loading` from `useCurrentProject` is `false` initially. 

| | |
|--------|--------|
| <img width="525" alt="Screenshot 2023-09-13 at 13 23 45" src="https://github.com/gitpod-io/gitpod/assets/914497/25a821c4-e769-4ee5-b77f-4ae9707d43fb"> | <img width="429" alt="Screenshot 2023-09-13 at 13 23 53" src="https://github.com/gitpod-io/gitpod/assets/914497/76b0f166-1a36-4280-a538-e233e3896ef5"> | 

I rely on the `isLoading` state existing in `<Projects>` to fix this, but it seems to be a the same case in other places
<img width="460" alt="Screenshot 2023-09-13 at 13 31 23" src="https://github.com/gitpod-io/gitpod/assets/914497/4f29ff9a-2075-4afa-af8c-83a782ff3828">
Therefore, I'll go with a partial revert in this hook. Maybe we can improve this later on, WDYT?

Reply:
@selfcontained, I think this might happen because we also did add `enabled: !!orgId` to `useListProjectsQuery`. The query isn't in loading state if it's not enabled, is it?

Reply:
I was testing this a few times by now, and it works nicely so far, thus considering this as resolved.